core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala

```diff
@@ -124,8 +124,11 @@ extends ReadablePartition with MapHolder {
     for { col <- 0 until numColumns optimized } {
       currentChunks(col).addFromReaderNoNA(row, col) match {
         case r: VectorTooSmall =>
-          switchBuffers(blockHolder, encode=true)
-          ingest(row, blockHolder)   // re-ingest every element, allocating new WriteBuffers
+          // NOTE: a very bad infinite loop is possible if switching buffers fails (if the # rows is 0) but one of the
+          // vectors fills up.  This is possible if one vector fills up but the other one does not for some reason.
+          // So we do not call ingest again unless switcing buffers succeeds.
+          // re-ingest every element, allocating new WriteBuffers
+          if (switchBuffers(blockHolder, encode=true)) ingest(row, blockHolder)
```

Comment:
The only way for switchBuffers to return false is if the write buffers are empty - which should never be the case, since the case is for detecting when write buffers are full.  This was an extreme corner case created by a bad bug in the early histogram write buffer.  Basically if this returns false there is something really bad going on.   

I'm going to add a giant WARN here.